pkg/proxy/service.go

```diff
@@ -278,15 +278,15 @@ type UpdateServiceMapResult struct {
 	UDPStaleClusterIP sets.String
 }
 
-// UpdateServiceMap updates ServiceMap based on the given changes.
-func UpdateServiceMap(serviceMap ServiceMap, changes *ServiceChangeTracker) (result UpdateServiceMapResult) {
+// Update updates ServiceMap based on the given changes.
```

Comment:
If we're going to do this, we should move the `type` definition above the method.

Reply:
Thanks for your guide @thockin . I found a similar problem in `pkg/proxy/endpoints.go`.
So I tried to adjust the relative position of the `type` definition and its methods.
IMO, this can improve the readability of our code. 
You are welcome to have another look :)